frontend/src/metabase/visualizations/visualizations/Table.jsx

```diff
@@ -185,7 +192,9 @@ export default class Table extends Component {
         cols.map(col => ({
           name: col.name,
           fieldRef: col.field_ref,
-          enabled: col.visibility_type !== "details-only",
+          enabled:
+            this.props.datasetEditorTab === "metadata" ||
```

Comment:
This seems to crash the app. As `settings` is static, so it has no access to table instances. i.e. `this.props` is `undefined` here. I'm not sure how to get around this yet.

Reply:
Right, makes sense. In that case, I need to repeat the same predicate in another place though. 
https://github.com/metabase/metabase/blob/b8339af7c629757dbe2e13e2a956ef442536ac69/frontend/src/metabase/visualizations/visualizations/Table.jsx#L404

Do you think it would be better to extract this predicate as a separate function?

Reply:
> Do you think it would be better to extract this predicate as a separate function?

Good catch! I didn't realize it's used in other places too. Yeah let's extract the predicate into its own function so it could be reused.

Reply:
Actually, I realised I can replace that second predicate by looking to see if the columns in `data` are empty. So no need to extract it.

https://github.com/metabase/metabase/blob/cbf80a90d51970b23c7b4fd5971719ba631d1cfe/frontend/src/metabase/visualizations/visualizations/Table.jsx#L404